### PR TITLE
FYST-1091-update-copy-on-nc-refund-debit-screen

### PR DESCRIPTION
**Pull request by DrewProebstel**
## Link to pivotal/JIRA issue
- https://codeforamerica.atlassian.net/browse/FYST-1209
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- Creates a new NC taxes owed page
- creates NC tax refund page
- updates radio button lay out in bank details partial
## How to test?
- heroku
## Screenshots (for visual changes)
Screenshot 2024-12-03 at 10 23 09 AM
Screenshot 2024-12-03 at 10 24 52 AM
Screenshot 2024-12-03 at 10 25 24 AM

Screenshot 2024-12-03 at 12 01 58 PM

### CODE REVIEW

File: app/views/state_file/questions/nc_taxes_owed/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            ) %>
+      </div>
+      <div class="question-with-follow-up__follow-up" id="pay-from-bank">
+        <div class="question-with-follow-up">
+          <div class="question-with-follow-up__question">
+            <%= render 'state_file/questions/nc_tax_refund/nc_bank_details', form: f, owe_taxes: true %>
+            <div id="details" class="reveal">
+              <p><button class="reveal__button"><%= t(".more_info_about_automatic_draft_payments") %></button></p>
+              <div class="reveal__content">
+                <p> <%= t(".automatic_draft_payments_info_one") %> </p>
```
Comment (embarnard) [inline app/views/state_file/questions/nc_taxes_owed/edit.html.erb:36]: such a nit pick but are these extra spaces intentional? Also on lines 43-45 and line 56 [dust]

File: app/views/state_file/questions/tax_refund/_bank_details.html.erb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <div class="white-group">
   <p class="spacing-below-0"><%= t(".bank_title") %></p>
   <p class="text--small"><%= t(".foreign_accounts") %></p>
-  <div class="form-group-tight">
+  <div>
```
Comment (embarnard) [inline app/views/state_file/questions/tax_refund/_bank_details.html.erb:4]: just out of curiosity why was this removed?

File: spec/factories/state_file_nc_intakes.rb [hunk 3]
```diff
@@ -120,6 +120,19 @@
       after(:create, &:synchronize_df_w2s_to_database)
     end
 
+    trait :taxes_owed do
```
Comment (embarnard) [inline spec/factories/state_file_nc_intakes.rb:123]: ooo nice i love new traits

File: config/locales/en.yml [hunk 4]
```diff
@@ -2868,6 +2868,41 @@ en:
...
+          direct_debit_payment_info_two: The draft date is based on a 5:00 pm EST cut off. In other words, if you submit your bank draft payment after 5:00 pm EST, the earliest draft date available will be two business days in the future.
+          how_would_you_like_to_pay: How would you like to pay?
+          more_info_about_automatic_draft_payments: More information about automatic draft payments
+          more_info_about_direct_debit_payment: More information about the direct debit payment date
+          page_title: You owe state taxes. How would you like to pay?
+          pay_bank: Pay directly from your checking or savings account (direct debit)
+          pay_mail_online_html: Pay by mail or online at <a target="_blank" rel="noopener nofollow" href="https://electronic-services.dor.nc.gov/wps/portal/d400v">NCDOR.gov</a>
+          payment_options: You can pay directly from your checking or savings account here. You can also pay by mail or online through the North Carolina Department of Revenue’s website.
+          title_html: You owe <strong>$%{owed_amount}</strong> in %{state_name} state taxes.
```
Comment (embarnard) [inline config/locales/en.yml:2904]: should we remove this `%{state_name}` interpolation now that this is only getting used in North Carolina? [dust]

File: spec/features/state_file/complete_intake_spec.rb [hunk 5]
```diff
@@ -372,7 +372,7 @@
       expect(page).to have_text I18n.t("state_file.questions.shared.review_header.title")
       click_on I18n.t("general.continue")
 
-      expect(strip_html_tags(page.body)).to include strip_html_tags(I18n.t("state_file.questions.tax_refund.edit.title_html", refund_amount: 1000, state_name: "North Carolina"))
+      expect(strip_html_tags(page.body)).to include strip_html_tags(I18n.t("state_file.questions.nc_tax_refund.edit.title_html", refund_amount: 1000, state_name: "North Carolina"))
```
Comment (embarnard) [inline spec/features/state_file/complete_intake_spec.rb:375]: remove `state_name: "North Carolina"` if change in yml above

File: app/controllers/state_file/questions/nc_tax_refund_controller.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+module StateFile
+  module Questions
+    class NcTaxRefundController < QuestionsController
```
Comment (embarnard) [inline app/controllers/state_file/questions/nc_tax_refund_controller.rb:3]: if these are the same as `TaxRefundController` except for the form key we could have it inherit from that controller and remove the other methods and just have the form_key method in here [dust]
```suggestion
    class NcTaxRefundController < TaxRefundController
```

File: app/controllers/state_file/questions/nc_taxes_owed_controller.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
+module StateFile
+  module Questions
+    class NcTaxesOwedController < QuestionsController
```
Comment (embarnard) [inline app/controllers/state_file/questions/nc_taxes_owed_controller.rb:3]: same as above [dust]
```suggestion
    class NcTaxesOwedController < TaxesOwedController
```

**Review by embarnard — APPROVED** [review]
Looks really good! just a few minor comments

File: app/views/state_file/questions/tax_refund/_bank_details.html.erb (hunk 2 above)
Comment (DrewProebstel) [reply app/views/state_file/questions/tax_refund/_bank_details.html.erb:4]: it makes the bullet points vertical which is more aesthetic
